packages/wallets/torus/src/adapter.ts

```diff
@@ -1,109 +1,98 @@
+        // this.ready = true;
     }
 
     get publicKey(): PublicKey | null {
-        return this._keypair?.publicKey || null;
+        return this._publicKey;
     }
 
     get ready(): boolean {
-        return typeof window !== 'undefined';
+        return true
     }
 
     get connecting(): boolean {
         return this._connecting;
     }
 
     get connected(): boolean {
-        return !!this._keypair;
+        return !!this._torus?.isLoggedIn;
     }
 
     async connect(): Promise<void> {
         try {
             if (this.connected || this.connecting) return;
             this._connecting = true;
 
-            let openLogin: OpenLogin;
-            let privateKey: string;
+            // Check if torus is initialized, torus.init({config})
+            let torus = typeof window !== 'undefined' && window.torus;
+            if (!torus) torus = new Torus();
+            window.torus = torus;
+            if (!torus.isInitialized) await torus.init(this._config);
+
+            // Login
+            let loginResult;
             try {
-                openLogin = new OpenLogin(this._options);
-
-                await openLogin.init();
-
-                privateKey = openLogin.privKey;
-                if (!privateKey) {
-                    let listener: (event: { reason: any }) => void;
-                    try {
-                        privateKey = await new Promise((resolve, reject) => {
-                            listener = ({ reason }) => {
-                                switch (reason?.message.toLowerCase()) {
-                                    case 'user closed popup':
-                                        reason = new WalletWindowClosedError(reason.message, reason);
-                                        break;
-                                    case 'unable to open window':
-                                        reason = new WalletWindowBlockedError(reason.message, reason);
-                                        break;
-                                }
-                                reject(reason);
-                            };
-
-                            window.addEventListener('unhandledrejection', listener);
-
-                            openLogin.login().then(
-                                // HACK: result.privKey is not padded to 64 bytes, use provider.privKey
-                                (result) => resolve(openLogin.privKey),
-                                (reason) => listener({ reason })
-                            );
-                        });
-                    } finally {
-                        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-                        window.removeEventListener('unhandledrejection', listener!);
-                    }
-                }
+                loginResult = await torus.login();
             } catch (error: any) {
                 if (error instanceof WalletError) throw error;
```

Comment:
This check is no longer necessary -- `torus.login()` shouldn't throw `WalletError`.